scripts/project.js

```diff
@@ -1,36 +1,50 @@
-var projects = [];
+(function(module) {
 
-function Project(projObj) {
-  this.title = projObj.title;
-  this.imgsrc = projObj.imgsrc;
-  this.caption = projObj.caption;
-  this.date = projObj.date;
-};
+  function Project(projObj) {
+    for (var key in projObj) {
+      this[key] = projObj[key];
+    }
+    for (var key in this.code) {
+      this.codehtml = this.codehtml + '<p>'+ key + ': ' + this.code[key] + ' lines</p>';
+    }
+  };
 
-Project.prototype.toHTML = function() {
-  var template = $('#project-template').html();
-  var templateRender = Handlebars.compile(template);
-  return templateRender(this);
-};
+  Project.projects = [];
 
-projectData.forEach(function(proj) {
-  projects.push(new Project(proj));
-});
+  Project.prototype.toHTML = function() {
+    var template = $('#project-template').html();
+    var templateRender = Handlebars.compile(template);
+    return templateRender(this);
+  };
 
-$projects = $('.project-carousel');
+  Project.loadAll = function(inputData) {
+    inputData.forEach(function(proj) {
+      Project.projects.push(new Project(proj));
+    });
+  };
 
-projects.forEach(function(proj) {
-  $projects.append(proj.toHTML());
-});
+  Project.fetchAll = function() {
+    $.ajax({
+      type: 'HEAD',
+      url: '../projects.json',
+      success: function(data, message, xhr) {
+        if (!localStorage.eTag || localStorage.eTag !== xhr.getResponseHeader('ETag')) {
+          console.log('load new data');
+          $.getJSON('../projects.json', function(data) {
+            localStorage.setItem('eTag', xhr.getResponseHeader('ETag'));
+            localStorage.setItem('projectData', JSON.stringify(data));
+            Project.loadAll(data);
+            viewController.renderProjects();
+          });
+        } else {
+          console.log('load from local');
+          Project.loadAll(JSON.parse(localStorage.getItem('projectData')));
+          viewController.renderProjects();
+        }
+      }
+    });
+  };
```

Comment:
TODO: add Article method/s to compute some statistics pertaining to the lines of code written in my projects. The "code" JSON property will have to be changed to an array, either in the JSON itself or with a function. That array can then be mapped and reduced.